src/webgpu/api/validation/queue/copyToTexture/CopyExternalImageToTexture.spec.ts

```diff
@@ -425,26 +425,35 @@ g.test('source_imageBitmap,state')
 g.test('source_canvas,state')
   .desc(
     `
-  Test HTMLCanvasElement as source image in state [placeholder, valid].
+  Test HTMLCanvasElement as source image in state [nocontext, placeholder, valid].
+
+  Nocontext means using a canvas without any context as copy param.
 
   Call 'transferControlToOffscreen' on HTMLCanvasElement will cause the
   canvas control right transfer. And this canvas is in state 'placeholder'
+  Whether getContext in new generated offscreenCanvas won't affect the origin
+  canvas state.
   
+  
+  Check whether 'OperationError' is generated when HTMLCanvasElement has no
+  context.
+
   Check whether 'InvalidStateError' is generated when HTMLCanvasElement is
   in 'placeholder' state.
     `
   )
   .params(u =>
     u //
-      .combine('state', ['placeholder', 'valid'])
+      .combine('state', ['nocontext', 'placeholder', 'valid'])
       .beginSubcases()
+      .combine('getContextInOffscreenCanvas', [false, true])
```

Comment:
nit: This generates 6 subcases but really there are only 4 different ones, I think. How about just `.combine('state', ['nocontext', 'placeholder-nocontext', 'placeholder-hascontext', 'valid'])`?